internal/service/FileCategories.go

```diff
@@ -710,47 +710,53 @@ var MimeToCategory = map[string]model.Category{
-	"application/vnd.oasis.opendocument.spreadsheet":  model.CategoryDocument,
-	"application/vnd.oasis.opendocument.text":         model.CategoryDocument,
-	"application/x-7z-compressed":                     model.CategoryArchive,
-	"application/x-rar-compressed":                    model.CategoryArchive,
-	"application/x-tar":                               model.CategoryArchive,
-	"application/zip":                                 model.CategoryArchive,
+	"application/gzip":             model.CategoryArchive,
+	"application/octet-stream":     model.CategoryBinary,
+	"application/epub+zip":         model.CategoryDocument,
+	"application/msword":           model.CategoryDocument,
+	"application/pdf":              model.CategoryDocument,
+	"application/x-7z-compressed":  model.CategoryArchive,
+	"application/x-rar-compressed": model.CategoryArchive,
+	"application/x-tar":            model.CategoryArchive,
+	"application/zip":              model.CategoryArchive,
+}
 
-	// office formats
-	"application/vnd.openxmlformats-officedocument.wordprocessingml.document":   model.CategoryDocument,
-	"application/vnd.openxmlformats-officedocument.wordprocessingml.template":   model.CategoryDocument,
-	"application/vnd.openxmlformats-officedocument.presentationml.template":     model.CategoryDocument,
-	"application/vnd.openxmlformats-officedocument.presentationml.slideshow":    model.CategoryDocument,
-	"application/vnd.openxmlformats-officedocument.presentationml.presentation": model.CategoryDocument,
-	"application/vnd.openxmlformats-officedocument.presentationml.slide":        model.CategoryDocument,
-	"application/vnd.openxmlformats-officedocument.spreadsheetml.sheet":         model.CategoryDocument,
-	"application/vnd.openxmlformats-officedocument.spreadsheetml.template":      model.CategoryDocument,
+var MimePrefixToCategory = map[string]model.Category{
+	"application/vnd.ms-excel":                      model.CategoryDocument,
+	"application/vnd.ms-word":                       model.CategoryDocument,
+	"application/vnd.oasis.opendocument":            model.CategoryDocument,
+	"application/vnd.openxmlformats-officedocument": model.CategoryDocument,
 }
 
 func GetFileCategory(name string, mimeType string, isDir bool) (model.Category, string) {
 	if isDir {
 		return model.CategoryFolder, ""
 	}
 	if mimeType == "" || mimeType == "application/octet-stream" {
-		if trueMimeType, ok := ExtToMime[filepath.Ext(name)]; ok {
+		if trueMimeType, ok := ExtToMime[strings.ToLower(filepath.Ext(name))]; ok {
 			mimeType = trueMimeType
 		}
 	}
 
+	// Check if the mime type matches a known category.
 	if cat, ok := MimeToCategory[mimeType]; ok {
 		return cat, mimeType
 	}
+
+	// Check if the mime type category matches.
 	slash := strings.IndexRune(mimeType, '/')
 	if slash > 0 {
 		if cat, ok := MimeToCategory[mimeType[0:slash]]; ok {
 			return cat, mimeType
 		}
 	}
+
+	// Check if the mime type has a known prefix.
+	for prefix, cat := range MimePrefixToCategory {
```

Comment:
The code might look a bit nicer but this iteration is going to get very slow and called very often. Is it worth it?

Reply:
Now there's only a few mimetypes in there, but that won't stay like that I think?

Reply:
The dots are part of the mimetype specifictation so maybe a better pattern would be: 

- Check prefix for mimeType[0:lastIndexOfDot]
- Strip a bit more from the end
- Repeat until match or no more dots?

Then you can use hash lookups again.

Reply:
Good points, I agree the iteration might be too much, especially for large folders. I like the dots idea though, that would be at most a couple of extra has lookups per mime type rather than iterating over the entire map so way better.

Reply:
Should be better now! I also moved the prefixes back in the `MimeToCategory` map since with this logic it doesn't make sense to keep them separate anymore.